app/jobs/discovery_report_job.rb

```diff
@@ -6,10 +6,15 @@ class DiscoveryReportJob < ApplicationJob
   # @param [JobRun] job_run
   def perform(job_run)
     job_run.started
-    report = job_run.to_discovery_report
-    file = File.open(report.output_path, 'w') { |f| f << report.to_builder.target! }
-    job_run.output_location = file.path # don't call report.output_path again
-    job_run.save!
-    job_run.completed
+    begin
+      report = job_run.to_discovery_report
+      file = File.open(report.output_path, 'w') { |f| f << report.to_builder.target! }
+      job_run.output_location = file.path # don't call report.output_path again
+      job_run.save!
+      job_run.completed
+    rescue StandardError => e
+      job_run.error_message = e.message
+      job_run.failed
+    end
   end
```

Comment:
We will now rescue any overall job exceptions and put the job into a failed state with the error message we can show to the user.  We can use this later to raise exceptions for things like bad manifests and have them caught here and reported to the user.